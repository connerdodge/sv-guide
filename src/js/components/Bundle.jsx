import React from 'react';
import styled from 'styled-components';

import ListItem from './ListItem';

function Bundle({ name, icon, reward, items, season }) {
  const filteredItems = items
    .filter(item => [season, 'any'].includes(item.season))
    .map(item => (<ListItem key={item.itemName} item={item}></ListItem>));

  if (!filteredItems.length) return null;

  return (
    <Container>
      <StyledName>{name}</StyledName>
      <StyledInfo>Reward: {reward}</StyledInfo>
      <StyledItemContainer>{filteredItems}</StyledItemContainer>
    </Container>
  );
}

const Container = styled.div`
  margin-top: 34px;
  color: #444;
`;

const StyledName = styled.span`
  font-weight: bold;
  padding-right: 10px;
`;

const StyledInfo = styled.span`
  padding-right: 10px;
`;

const StyledItemContainer = styled.div`
  display: flex;
  flex-direction: column;
  margin: 0 auto;
  width: fit-content;
`;

export default Bundle;
