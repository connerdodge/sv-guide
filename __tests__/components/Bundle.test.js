import React from 'react';
import { render } from '@testing-library/react';

import Bundle from '../../src/js/components/Bundle';

const defaultProps = {
  season: 'spring',
  name: 'Spring Foraging',
  icon: 'https://stardewvalleywiki.com/mediawiki/images/thumb/b/b3/Bundle_Green.png/32px-Bundle_Green.png',
  reward: 'Spring Seeds (30)',
  items: [
    {
      season: 'spring',
      itemName: 'Wild Horseradish',
      icon: 'https://stardewvalleywiki.com/mediawiki/images/thumb/9/90/Wild_Horseradish.png/24px-Wild_Horseradish.png',
      location: [],
      time: '',
      weather: ''
    },
  ],
};

it('renders and matches snapshot', () => {
  const { container } = render(<Bundle {...defaultProps} />);
  expect(container).toMatchSnapshot();
});
