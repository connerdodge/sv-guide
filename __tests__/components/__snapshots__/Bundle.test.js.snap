// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and matches snapshot 1`] = `
<div>
  <div
    class="sc-hKgJUU btQklQ"
  >
    <span
      class="sc-eCstlR bXLgGK"
    >
      Spring Foraging
    </span>
    <span
      class="sc-jSgvzq gopgTe"
    >
      Reward: 
      Spring Seeds (30)
    </span>
    <div
      class="sc-gKseQn bPHTMe"
    >
      <div
        class="sc-bdfBQB cnnJGR"
      >
        <input
          type="checkbox"
          class="sc-gsTEea fmJIQB"
        />
        <span
          class="sc-dlfnuX pUKlA"
        >
          Wild Horseradish
        </span>
      </div>
    </div>
  </div>
</div>
`;
